Source_Code/TM_UnitTests/TeamMentor.UnitTests/TeamMentor.UserData/Users/Test_Users_Creation.cs

```diff
@@ -157,6 +157,93 @@ public void CreateTmUserSigupResponse_Success()
         }
 
         [Test]
+        public void CreateTmUserSigupResponse_TLD_Email_Success()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+
+            var newUser = new NewUser()
+            {
+                Company = 10.randomLetters(),
+                Country = 10.randomLetters(),
+                Firstname = 10.randomLetters(),
+                Lastname = 10.randomLetters(),
+                Note = 10.randomLetters(),
+                Password = "!r4J" + 10.randomLetters(),
+                State = 10.randomLetters(),
+                Title = 10.randomLetters(),
+                Username = 10.randomLetters(),
+                Email = "dmin@xn--xn--fsqu00a.xn--g6w251d"
+            };
+
+            Assert.IsEmpty(newUser.validate());
+
+            var response = newUser.createSigupResponse();
+            Assert.IsTrue(response.notNull());
+            Assert.IsTrue(response.UserCreated > 0);
+            Assert.IsTrue(response.Signup_Status == Signup_Result.SignupStatus.Signup_Ok);
+            Assert.IsTrue(response.Validation_Results.count() == 0);
+
+        }
+
+        [Test]
+        public void CreateTmUserSigupResponse_EmailMaxLength_Allowed()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+
+            var newUser = new NewUser()
+            {
+                Company = 10.randomLetters(),
+                Country = 10.randomLetters(),
+                Firstname = 10.randomLetters(),
+                Lastname = 10.randomLetters(),
+                Note = 10.randomLetters(),
+                Password = "!r4J" + 10.randomLetters(),
+                State = 10.randomLetters(),
+                Title = 10.randomLetters(),
+                Username = 10.randomLetters(),
+                Email = "".random_Email() + "".add_RandomLetters(230)
+            };
+             
+            Assert.IsTrue(newUser.Email.Length ==256);

```

Comment:
can use use the FluentSharp assert methods so that it reads:

```csharp
newUser.Email.Length.assert_Is(256);
//or
newUser.Email.assert_Size_Is(256);
```